dask/dataframe/io/parquet/arrow.py

```diff
@@ -89,19 +89,19 @@ def read_metadata(
                     dataset.metadata.row_group(i)
                     for i in range(dataset.metadata.num_row_groups)
                 ]
-                # TODO get names
-                # names = schema.names  # Not quite right when partitioning
+                names = dataset.metadata.schema.names
             else:
                 # Read from each individual piece (quite possibly slow)
                 row_groups = [
-                    piece.get_metadata(
-                        lambda fn: pq.ParquetFile(fs.open(fn, mode="rb"))
```

Comment:
I think that the `fs.open` bit is here to help support file systems that are different than the local Posix system.  This might be something like S3, GCS, or HDFS.

Reply:
This might no longer be necessary, I don't know, but I suspect that there is some Parquet + S3 test already in the test suite to help verify.

Reply:
Right, we *shouldn't* need this anymore, because  the `open_file_func` is set by the `ParquetDataset` constructor ([in here](https://github.com/apache/arrow/blob/740e7293c885a96d4a79637d0e984422f2b3fe64/python/pyarrow/parquet.py#L897)).  However, since this change was just to avoid a deprecation warning, we can easily revert if needed.